src/tests/suites/KubectlClientProvider-Test.ts

```diff
@@ -0,0 +1,94 @@
+import { expect } from 'chai';
+import { after, describe, it } from 'mocha';
+import * as process from 'process';
+import sinon = require('sinon');
+import { CommandRunner } from '../../clients/CommandRunner';
+import { IClientProvider } from '../../clients/Providers/IClientProvider';
+import { accountContextManagerStub, loggerStub } from '../CommonTestObjects';
+import { BinariesVersionClient } from '../../clients/BinariesVersionClient';
+import { KubectlClientProvider } from '../../clients/Providers/KubectlClientProvider';
+
+describe('KubectlClientProviderTest', () => {
+    const originalPlatform = Object.getOwnPropertyDescriptor(process, 'platform');
+    it('should always return linux/kubectl for linux', async () => {
+        Object.defineProperty(process, 'platform', {
+            value: 'linux'
+        });
+        const expectedCLIVersion = '1.0.20220816.2';
```

Comment:
💡 Just an idea - but see all the repetitive code to instantiate the `BrnariesVersionClient and then createSubInstances` we could private funct and call to avoid repetitive things. Given its tiny, I will leave that as a decision for author to decide :) rest looks cool, thank you so much!